src/Coderockr/Mongodb/ServiceProvider.php

```diff
@@ -8,21 +8,58 @@
 
 class ServiceProvider implements ServiceProviderInterface
 {
-    private $options = [
-        'uri' => "mongodb://localhost:27017",
-        'options' => [],
-        'driverOptions' => []
-    ];
-
-    public function __construct(array $options)
-    {
-        $this->options = array_merge($this->options, $options);
-    }
-
     public function register(Application $app)
     {
-        $app['mongodb.manager'] = $app->share(function () {
-            return new Manager($this->options['uri'], $this->options['options'], $this->options['driverOptions']);
+        $app['mongodbs.options.initializer'] = $app->protect(function () use ($app) {
+            static $initialized = false;
+
+            if ($initialized) {
+                return;
+            }
+
+            $initialized = true;
+
+            if (!isset($app['mongodbs.options'])) {
+                $app['mongodbs.options'] = [
+                    'default' => isset($app['mongodb.options']) ? $app['mongodb.options'] : []
+                ];
+            }
+
+            $tmp = [];
+            foreach ($app['mongodbs.options'] as $name => $options) {
+                $tmp[$name] = array_replace($app['mongodb.default_options'], $options);
+
+                if (!isset($app['mongodbs.default'])) {
+                    $app['mongodbs.default'] = $name;
+                }
+            }
+
+            $app['mongodbs.options'] = $tmp;
+        });
```

Comment:
Então, tentei ir por esse caminho, mas o php lança: `Indirect modification of overloaded element`